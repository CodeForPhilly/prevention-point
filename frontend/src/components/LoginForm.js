import React, { useState } from "react"
import PropTypes from "prop-types"
import { Redirect } from "react-router-dom"
import { observer } from "mobx-react-lite"

import "../scss/login-form.scss"
import FormGroup from "@material-ui/core/FormGroup"
import FormControl from "@material-ui/core/FormControl"
import InputLabel from "@material-ui/core/InputLabel"
import Input from "@material-ui/core/Input"
import Button from "@material-ui/core/Button"
import api from "../api"
import { useContextAuth, LOGIN } from "../contexts/auth"

const LoginForm = observer(({ location }) => {
  const [username, setUsername] = useState("")
  const [password, setPassword] = useState("")
  const { pathname } = location.state.from || { pathname: "/" }

  const changeUsername = e => setUsername(e.target.value)

  const changePassword = e => setPassword(e.target.value)

  const {
    auth: { username: authUsername },
    dispatchAuth,
  } = useContextAuth()
  const login = async () => {
    try {
      const { ok } = await api.createToken(username, password)
      if (ok) dispatchAuth(LOGIN(username))
    } catch (error) {
      // TODO: Handle errors
    }
  }
  if (authUsername) return <Redirect to={pathname} />

  return (
    <div className="login-form">
      <form className="login-form__form">
        <FormGroup className="login-form__input">
          <FormControl>
            <InputLabel htmlFor="username">Username</InputLabel>
            <Input
              id="username"
              name="username"
              type="username"
              value={username}
              onChange={changeUsername}
              required
            />
          </FormControl>
        </FormGroup>
        <FormGroup className="login-form__input">
          <FormControl>
            <InputLabel htmlFor="password">Password</InputLabel>
            <Input
              id="password"
              name="password"
              value={password}
              type="password"
              onChange={changePassword}
              required
            />
          </FormControl>
        </FormGroup>
        <Button
          type="button"
          variant="contained"
          style={{ marginTop: "10px" }}
          onClick={login}
        >
          Sign In
        </Button>
      </form>
    </div>
  )
})

LoginForm.propTypes = {
  location: PropTypes.object,
}

export default LoginForm
