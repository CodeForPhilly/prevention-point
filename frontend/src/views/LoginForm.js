import React, { useState, useContext } from "react"
import PropTypes from "prop-types"
import { observer } from "mobx-react-lite"
import { Redirect } from "react-router-dom"
import { makeStyles } from "@material-ui/core/styles"
import PreventionPointLogo from "../../public/img/logo.svg"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import Grid from "@material-ui/core/Grid"
import Container from "@material-ui/core/Container"
import InputLabel from "@material-ui/core/InputLabel"
import FormControl from "@material-ui/core/FormControl"
import PrevPointInput from "../components/Input/PrevPointInput"
import PrevPointButton from "../components/PrevPointButton"
import PrevPointCopy from "../components/Typography/PrevPointCopy"
import { rootStoreContext } from "../stores/RootStore"

const LoginForm = observer(({ location }) => {
  const rootStore = useContext(rootStoreContext)
  const [username, setUsername] = useState("")
  const [password, setPassword] = useState("")
  const { from } = location.state || { from: { pathname: "/" } }

  const useStyles = makeStyles(theme => ({
    root: {
      padding: theme.spacing(6, 0, 0, 0),
      "& .login-form__form": {
        margin: "0 auto",
        maxWidth: "400px",
      },
      "& .login-form__error": {
        color: "#ff000f",
        paddingTop: theme.spacing(2),
      },
    },
    appBar: {
      "& .navbar__logo": {
        marginRight: "auto",
      },
    },
    navbarToolbar: {
      justifyContent: "flex-end",
      paddingTop: "8px",
      paddingBottom: "8px",
      "@media (max-width: 700px)": {
        flexWrap: "wrap",
      },
    },
    navbarOffset: {
      ...theme.mixins.toolbar,
      "@media (max-width: 700px)": {
        minHeight: "116px",
      },
    },
  }))
  const classes = useStyles()

  const login = event => {
    event.preventDefault()
    rootStore.authStore.login(username, password)
  }

  if (rootStore.authStore.isAuthenticated) return <Redirect to={from} />

  return (
    <>
      <AppBar
        title="Prevention Point"
        position="fixed"
        className={classes.appBar}
      >
        <Toolbar className={classes.navbarToolbar}>
          <img
            src={PreventionPointLogo}
            alt="Prevention Point Logo"
            className="navbar__logo"
            width="150"
          />
        </Toolbar>
      </AppBar>

      <div className={classes.navbarOffset} />
      <Container className={classes.root}>
        <Grid
          container
          component="form"
          className="login-form__form"
          onSubmit={login}
        >
          <Grid item xs={12}>
            <FormControl error={rootStore.authStore.error}>
              <InputLabel htmlFor="username">Username</InputLabel>
              <PrevPointInput
                id="username"
                name="username"
                type="username"
                value={username}
                onChange={e => setUsername(e.target.value)}
                required
              />
            </FormControl>
          </Grid>
          <Grid item xs={12}>
            <FormControl error={rootStore.authStore.error}>
              <InputLabel htmlFor="password">Password</InputLabel>
              <PrevPointInput
                id="password"
                name="password"
                type="password"
                value={password}
                onChange={e => setPassword(e.target.value)}
                required
              />
            </FormControl>
          </Grid>
          <Grid item xs={12}>
            <PrevPointButton type="submit">Sign In</PrevPointButton>
            {rootStore.authStore.error && (
              <PrevPointCopy className="login-form__error">
                Incorrect Username or password
              </PrevPointCopy>
            )}
          </Grid>
        </Grid>
      </Container>
    </>
  )
})

LoginForm.propTypes = {
  location: PropTypes.object,
}

export default LoginForm
