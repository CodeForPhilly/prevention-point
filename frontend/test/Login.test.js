import React from "react"
import { shallow } from "enzyme"
import LoginForm from "../src/components/LoginForm"

describe("<LoginForm /> with no props", () => {
  const container = shallow(<LoginForm />)
  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot()
  })

  it("should have a password field", () => {
    /* Similar to above */
  })
  it("should have proper props for password field", () => {
    /* Trimmed for less lines to read */
  })
  it("should have a submit button", () => {
    /* */
  })
  it("should have proper props for submit button", () => {
    /* */
  })
})
