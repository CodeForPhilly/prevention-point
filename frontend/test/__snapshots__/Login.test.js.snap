// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Unit tests should match the snapshot 1`] = `"<div class=\\"MuiContainer-root login-form MuiContainer-maxWidthLg\\"><form class=\\"MuiGrid-root login-form__form MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"username\\">Username</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><input type=\\"username\\" id=\\"username\\" name=\\"username\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"password\\">Password</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><input type=\\"password\\" id=\\"password\\" name=\\"password\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-PrevPointButton-168 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Sign In</span></button></div></form></div>"`;
